tractolearn/tractoio/dataset_fetch.py

```diff
@@ -0,0 +1,389 @@
+    hash_list=None,
+    doc="",
+    data_size=None,
+    msg=None,
+    unzip=False,
+):
+    """Create a new fetcher.
+
+    Parameters
+    ----------
+    folder : str
+        The full path to the folder in which the files would be placed locally.
+    name : str
+        The name of the fetcher function.
+    baseurl : str
+        The URL from which this fetcher reads files.
+    remote_fnames : list of strings
+        The names of the files in the baseurl location.
+    local_fnames : list of strings
+        The names of the files to be saved on the local filesystem.
+    hash_list : list of strings, optional
+        The hash values of the files. Used to verify the content of the files.
+        Default: None, skipping checking hash.
+    doc : str, optional.
+        Documentation of the fetcher.
+    data_size : str, optional.
+        If provided, is sent as a message to the user before downloading
+        starts.
+    msg : str, optional
+        A message to print to screen when fetching takes place. Default (None)
+        is to print nothing.
+    unzip : bool, optional
+        Whether to unzip the file(s) after downloading them. Supports zip, gz,
+        and tar.gz files.
+
+    Returns
+    -------
+    fetcher : function
+        A function that, when called, fetches data according to the designated
+        inputs
+
+    """
+
+    def fetcher():
+        files = {}
+        for (
+            i,
+            (f, n),
+        ) in enumerate(zip(remote_fnames, local_fnames)):
+            files[n] = (
+                baseurl + f,
+                hash_list[i] if hash_list is not None else None,
+            )
+        fetch_data(files, folder, data_size)
+
+        if msg is not None:
+            print(msg)
+        if unzip:
+            for f in local_fnames:
+                split_ext = os.path.splitext(f)
+                if split_ext[-1] == ".gz" or split_ext[-1] == ".bz2":
+                    if os.path.splitext(split_ext[0])[-1] == ".tar":
+                        ar = tarfile.open(pjoin(folder, f))
+                        ar.extractall(path=folder)
+                        ar.close()
+                    else:
+                        raise ValueError("File extension is not recognized")
+                elif split_ext[-1] == ".zip":
+                    z = zipfile.ZipFile(pjoin(folder, f), "r")
+                    files[f] += (tuple(z.namelist()),)
+                    z.extractall(folder)
+                    z.close()
+                else:
+                    raise ValueError("File extension is not recognized")
+
+        return files, folder
+
+    fetcher.__name__ = name
+    fetcher.__doc__ = doc
+    return fetcher
+
+
+fetch_contrastive_ae_weights = (
+    "fetch_contrastive_ae_weights",
+    TRACTOLEARN_DATASETS_URL + "2xmgw/",
+    ["download"],
+    ["weights.pt"],
+    ["7170d0192fa00b5ef069f8e7c274950c"],
+    "Download contrastive-loss trained autoencoder weights",
+    "543B",
```

Comment:
The data needs to be adjusted once it is uploaded to OSF.

Reply:
Done in [4bdf06b](https://github.com/scil-vital/tractolearn/commit/4bdf06b6316e08a540e14453ab68cc4136866f5c).